keps/1282-pods-ready-requeue-strategy/README.md

```diff
@@ -208,39 +209,37 @@ type RequeueState struct {
 }
 ```
 
-Add a new field `evictionStats` to the WorkloadStatus struct to track the number of evictions by type and reason:
+Add a new field, `evictionState`, to the WorkloadStatus struct to track the number of evictions by reason:
 
 ```go
 type WorkloadStatus struct {
-	// evictionStats tracks eviction statistics by type and reason.
+    // evictionState tracks eviction statistics by reason.
     //
     // +optional
     // +listType=map
-    // +listMapKey=type
+    // +listMapKey=reason
     // +patchStrategy=merge
-    // +patchMergeKey=type
-    EvictionStats []Eviction `json:"evictionStats,omitempty"`
+    // +patchMergeKey=reason
+    EvictionStates []WorkloadEvictionState `json:"evictionStates,omitempty"`
 }
 
-type Eviction struct {
-    // type identifies the eviction type.
+type WorkloadEvictionState struct {
```

Comment:
Eviction limits the applciation to eviction only. Can we make it a little bit more generic?

Reply:
We can try, but I'm worried we can over-engineer this. Do you have any specific proposal?

I think given the data is going to supply the EvictedWorkloadOnce metric it makes sense to keep the name scoped to "eviction".

We can always revisit the API for v1beta2.

Reply:
What I'm suggesting is just a minor rename. Instead of calling it eviction state maybe we can call it, for example, scheduling state - almost the same, but leaves other applications open.

Reply:
but it embeds the list of eviction reasons, directly, since this is `EvictionStates []WorkloadEvictionState`. 

Maybe we could add another layer of nesting to accommodate for it
```yaml
schedulingState:
  // with potential other info
  evictionState:
  - reason: ...
     detailedReason
     count: 
```

Reply:
OK, i understand calling the field evictionState and making it an array. What I'm trying to point out is that the structure has no field very specific to eviction it is reason, details and count. So I was thinking about removing eviction from the struct name to allow future reuse.

Reply:
I synced with @mwielgus and we concluded on the following option as preferred:

```yaml
schedulingState:
  // potential place to include admissionStats, etc.
  evictionStats:  
  - reason: PodsReadyTimeout  // From Evicted.reason
    detailedReason: WaitForStart  // From PodsReady.reason
    count: 3
```
but skipping `schedulingState` is also acceptable.